public/app/features/live/live.ts

```diff
@@ -113,51 +182,34 @@ class CentrifugeSrv implements GrafanaLiveSrv {
     return this.connectionState.asObservable();
   }
 
-  initChannel<T>(path: string, handler: ChannelHandler<T>) {
-    if (this.channels[path]) {
-      console.log('Already connected to:', path);
-      return;
-    }
-    const c: Channel = {
-      subject: new Subject<T>(),
-    };
-    this.channels[path] = c;
-
-    console.log('initChannel', this.centrifuge.isConnected(), path, handler);
-    const callbacks: SubscriptionEvents = {
-      ...this.standardCallbacks,
-      publish: (ctx: PublicationContext) => {
-        // console.log('GOT', JSON.stringify(ctx.data), ctx);
-        const v = handler.onPublish(ctx.data);
-        c.subject.next(v);
-      },
-    };
-    c.subscription = this.centrifuge.subscribe(path, callbacks);
-  }
-
-  getChannelStream<T>(path: string): Observable<T> {
-    let c = this.channels[path];
-    if (!c) {
-      this.initChannel(path, noopChannelHandler);
-      c = this.channels[path];
-    }
-    return c!.subject.asObservable();
+  getChannelStream<T>(plugin: string, path: string): Observable<T> {
+    return from(this.getChannel<T>(plugin, path)).pipe(
+      take(1), // This converts the promise to observable
+      mergeMap(c => c.stream)
+    );
   }
 
   /**
    * Send data to a channel.  This feature is disabled for most channels and will return an error
    */
-  publish<T>(channel: string, data: any): Promise<T> {
-    return this.centrifuge.publish(channel, data);
+  async publish<T>(plugin: string, path: string, data: any): Promise<T> {
+    try {
+      const support = await this.getChannelSupport(plugin);
+      if (!support.onPublish) {
+        return Promise.reject({
+          text: `Channel ${path} does not allow publishing`,
+        });
```

Comment:
Don't think async functions need to wrap return values or errors in promises, so the following could just be
```suggestion
        throw {
          text: `Channel ${path} does not allow publishing`,
        };
```